sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2189,4 +2189,50 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
           |*(1) Range (0, 10, step=1, splits=2)""".stripMargin))
     }
   }
+
+  test("SPARK-28445: PythonUDF in grouping key and aggregate expressions") {
```

Comment:
Shall we create like PythonUDFSuite and move those tests there? Not sure if `DataFrameSuite` is good for them.

Reply:
Yea, looks so. I think actually that's a good idea to have it.